packages/cli/src/project-references.ts

```diff
@@ -90,8 +95,10 @@ export function topologicalSort<Value>(
     recursionStack.add(node);
     visited.add(node);
 
-    const neighbours = graph.get(node);
-    assert(neighbours !== undefined);
+    // The graph only contains dependencies that are referenced in the parent
+    // `tsconfig.json`. If it's not referenced there, we can assume that it
+    // doesn't have any dependencies.
+    const neighbours = graph.get(node) ?? [];
```

Comment:
I'm running into the limits of my TypeScript knowledge here. In your tests for this, you use the following config:

```
    const graph: DependencyGraph<string> = new Map([
      ['a', ['b', 'd']],
      ['b', ['c', 'e']],
      ['c', ['d']],
      ['d', ['e']],
    ]);
```

In this setup, `e` has dependents but no dependencies. Does it only need to be listed in the root `tsconfig` references if it has dependencies, but not dependents? If that does not determine when it needs to be added as a reference, what does?

Reply:
If something has dependents, and thus is mentioned in one of the child `tsconfig.json`s, apparently it does not need to be in the root `tsconfig.json`. I discovered this when trying to build `MetaMask/core`, which is missing a package from the root, but it still builds that package.

I thought about a better error (the assert error was cryptic), but I want to match `tsc`'s behaviour as much as possible.